src/pybind/mgr/dashboard/controllers/user.py

```diff
@@ -79,6 +84,10 @@ def set(self, username, password=None, name=None, email=None, roles=None):
             user = mgr.ACCESS_CTRL_DB.get_user(username)
         except UserDoesNotExist:
             raise cherrypy.HTTPError(404)
+        if not password and not name and not email and (user.to_dict())['roles'] == roles:
```

Comment:
Is it not the same as this (I mean without the `to_dict()` step)?
```suggestion
        if not password and not name and not email and user.roles == roles:
```

Reply:
No, because user.roles contains the object and address, but to_dict() returns actual string value.